### PR TITLE
Improve legibility of registration links #5089 [TASK-1058] 

### LINKED ISSUE #5089: Make TOS and Privacy Policy links more legible on light backgrounds for registration page
## Description

The TOS and Privacy Policy links are not very legible on light backgrounds when seeing them on the registration/login page. 

## Steps to Reproduce

1. Load light or white background image on registration page
2. Visit registration page

## Expected behavior 

Links are easy to read. 

## Actual behavior 

Links are hard to read because they assume a dark background. 

## Additional details

Proposed solution: Move them into the dark div above so they can be read regardless of background.

### PR DESCRIPTION
**Pull request by tinok**
## Checklist

1. [ ] If you've added code that should be tested, add tests
2. [ ] If you've changed APIs, update (or create!) the documentation
3. [x] Ensure the tests pass
4. [x] Make sure that your code lints and that you've followed [our coding style](https://github.com/kobotoolbox/kpi/blob/master/CONTRIBUTING.md)
5. [x] Write a title and, if necessary, a description of your work suitable for publishing in our [release notes](https://community.kobotoolbox.org/tag/release-notes)
6. [x] Mention any related issues in this repository (as #ISSUE) and in other repositories (as kobotoolbox/other#ISSUE)
7. [ ] Open an issue in the [docs](https://github.com/kobotoolbox/docs/issues/new) if there are UI/UX changes

## Description

Move Terms of Service and Privacy Policy links inside the darker area to provide better visibility.

## Notes

Some changes in the UI were made based on [this discussion](https://chat.kobotoolbox.org/#narrow/stream/4-Kobo-Dev/topic/UI.2FUX.20check/near/457574) in Zulip.

| Production| Move links into dark shape<br> and add separator |
|-|-|
|image <br>(before this PR) |image <br>(before contrast revisions below)|

## Testing

- [x] Test wide and narrow layouts for signup page
- [x] Test with and without links configured in Constance settings
- [x] Test with and without SSO
  - xref [[internal doc for setting up test SSO](https://www.notion.so/kobotoolbox/Setting-up-SSO-3f85a206af1349dc8c4fc4c0c919e58e)]
  - [x] Test with SSO-specific login/signup flows

|                | No Links | With Links |
| -------------- | ----- | -------- |
| **Login**      | ✅🖼️  | ✅🖼️     |
| + SSO          | ✅🖼️  | ✅🖼️     |
| **Signup**     | ✅     | ✅        |
| + SSO          | ✅  | ✅🖼️        |
| + Narrow       | ✅     | ✅        |
| + SSO+Nar.     | ✅  | ✅🖼️        |
| **SSO login**  | ✅🖼️  | ✅🖼️        |
| **SSO signup** | ✅     | ✅        |

* *I included some screenshots of how this ended up; the rest of the screens look consistent with the ones shown.* — phil

### Basic Login Screenshots
|           | Without Links | With Links |
| --------- | -------- | ----- |
| **Login** | Screenshot_20241011_211321 | Screenshot_20241011_211459-1|
| + SSO     |  Screenshot_20241011_211305 | Screenshot_20241011_211251 |

### Signup Screenshots

| Narrow| Wide |
| --------- | -------- |
|Screenshot_20241011_211146|Screenshot_20241011_211123-1|

### SSO login
 | Without Links | With Links |
| ----- | -------- |
| Screenshot_20241011_193037 |   Screenshot_20241011_193419 |

## Related issues

Fixes #5089

### CODE REVIEW

**Review by p2edwards — COMMENTED** [review]
Cool! 

@pauloamorimbr  During review / edge case testing I made some suggested changes in a separate branch (https://github.com/kobotoolbox/kpi/tree/5089-fix-registration-links--additions):

- Checked what it looks like when an SSO provider is configured — I color-matched the SSO separator with this new links one (went with the new color)
- Fixed a layout issue on signup page on narrow screen width
  - Also fixed an unrelated color issue that was introduced recently — changed 'required' asterisks back to light red
- Remove extra separator if the Django admin didn't configure Privacy Policy / Terms links at all

Compare: https://github.com/kobotoolbox/kpi/compare/5089-fix-registration-links..5089-fix-registration-links--additions — (you'd need to merge 'main' into '5089-fix-registration-links' first to see a clean diff)

|  | +sso |
|-|-|
|image|Pasted image 20241004211441|


#### Other questions (not included above)

- :art: @tesster7 What do you think of these legal link colors/sizes/etc.? I wonder if we'd want them to match **more closely** with the Create Account / Forgot password or not. The contrast is still a little low imo but maybe want to avoid it being too busy.

#### **sso-specific pages**

- :wrench: @pauloamorimbr There are 2 more SSO-related templates (under kpi/kobo/apps/accounts/templates/socialaccount/ — login.html and signup.html) @tinok should we put the Terms / Privacy links here too, or is that not necessary? (the deep signup / login pages that you get if you're using SSO.)

| sso-specific login page | e.g. with legal links|
|-|-|
|image | Pasted image 20241004211246|

File: kobo/apps/accounts/templates/legal/registration_legal.html [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    </a>
+    {% endif %}
+    {% if config.PRIVACY_POLICY_URL %}
+    <a href="{{config.PRIVACY_POLICY_URL}}" target="_blank">
+      {% trans "Privacy Policy" %}
+    </a>
+    {% endif %}
+  </div>
+</div>
+{% endif %}
```
Comment (p2edwards) [inline kobo/apps/accounts/templates/legal/registration_legal.html:18]: :+1: :sunglasses::+1:   very cool!

File: kobo/apps/accounts/templates/account/signup.html [hunk 2]
```diff
@@ -91,6 +91,5 @@ <h2>{% trans "Register using SSO" %}</h2>
   </div>
   <div style="clear:both;"></div>
   {% include "../legal/registration_legal.html" with config=config %}
-</div>
```
Comment (p2edwards) [inline kobo/apps/accounts/templates/account/signup.html]: An extra `</div>` [snuck](https://github.com/kobotoolbox/kpi/pull/5090/commits/649d68caf4f1b5135fde43d04b1c4936fa9b3a2c#diff-316fadeed80679fe9a99ac2e78a3b2079a1de730ae3f9cd243af15677ca79e20R105) in here ([I removed it](https://github.com/kobotoolbox/kpi/pull/5090/commits/0fd278b6616e09fa44f8354ce5161748ed0872ad))

Was causing the photo credit to do this:


| before | after |
| ------ | ----- |
| Screenshot_20241011_185218| Screenshot_20241011_210602 |

**Review by p2edwards — APPROVED** [review]
:+1: :white_check_mark:  Ready to merge @pauloamorimbr — if you're cool with the pushed commits:

- I finished setting up SSO on my dev machine so I could check how signup looks. 
  - I adjusted it to match the others better, that's the only new thing
- Added a couple more fixes from the review branch (that got lost in the rebase I think)
  - I added screenshots in comments for this PR to make these easier to review
- Edited description with test plan and screenshots of how things look now
  - This change seemed simple. But with the SSO option, a responsive signup layout, and the Constance config options, it affected 4 templates and had at least 16 distinct variants. I listed them in the description in case we have to test something like this again.

**Review by pauloamorimbr — APPROVED** [review]
Great fixes.

**Pull request merged**